core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/mercury.go

```diff
@@ -86,7 +87,7 @@ type HttpClient interface {
 }
 
 type MercuryClient interface {
-	DoRequest(ctx context.Context, streamsLookup *StreamsLookup, pluginRetryKey string) (encoding.PipelineExecutionState, encoding.UpkeepFailureReason, [][]byte, bool, time.Duration, error)
+	DoRequest(ctx context.Context, streamsLookup *StreamsLookup, pluginRetryKey string) (encoding.PipelineExecutionState, encoding.UpkeepFailureReason, [][]byte, bool, time.Duration, encoding.ErrCode, error)
```

Comment:
Can we add a comment here on what all these fields are supposed to represent. 

It is confusing that we have multiple similar fields such as pipeline execution state, failure reason, error code and there are constraints that some of these things can't happen together. (e.g. if error code is returned then state should be no error etc)

Reply:
If retry interval returned here is `RetryIntervalTimeout`, then would we always expect a non zero error code?

Reply:
e.g. what would be the difference between error code and error